File: vscode/samurai-agent/tests/agent/code_parser/fixtures/python/sample_imports.py
# Python Imports Test Fixture  
# Tests import extraction including re-exports

# Regular imports
import os
import sys
from typing import List, Dict, Optional
from datetime import datetime

# Relative imports
from .models import User
from ..utils import helper

# Re-exports (Phase 4)
from .types import *
from .constants import API_KEY, MAX_RETRIES

# __all__ definition (controls what gets exported)
__all__ = ['process_user', 'User']

def process_user(user_id: str) -> Optional[dict]:
    return None

File: vscode/samurai-agent/tests/agent/code_parser/fixtures/typescript/sample_imports.ts
// TypeScript Imports Test Fixture
// Tests import extraction including type-only imports and re-exports

// Regular imports
import { User } from './types';
import * as utils from './utils';
import defaultExport from './default';

// Phase 4: Type-only imports
import type { UserId, UserRole } from './types';
import type { ApiResponse } from './api';

// Dynamic imports
const module = await import('./dynamic-module');

// Re-exports (Phase 4)
export { User } from './types';
export * from './utils';
export { default as MyClass } from './MyClass';

// Regular code
export function processUser(user: User): void {
    console.log(user);
}

